command-acceptor/app/controllers/implementations/CommandFactory.py

```diff
@@ -1,40 +1,74 @@
+        self._user_id = message.from_field.user_id  # should we validate id on here or just send it to the processor?
+        self._language = message.from_field.language_code
 
-    # def split_input(self) -> list:
-    #     return self._text.split()
+    def prepare_error_message(self, data_by_language: dict)-> str:
+        _error_message_text = 'The command "{}" has incorrect format. Command should contain an operation_name,  a subject_name and pointer. F.e. create file temp.txt. Available operations are: {}; availabel subjects are {}'
+        _operations = data_by_language['operation'].keys()
+        _subjects = data_by_language['operation'].keys()
+        _error_message_text.format(self._text, _operations, _subjects)
+        return _error_message_text
 
-    def command_values(self) -> dict:
+    def command_values(self) -> list:
         values_list = self._text.split()
-        values_dict = {'operation': values_list[0], 'subject':values_list[1], 'value':''} # Maybe None will be better than empty strng
-        if len(values_list) > 2:
-            values_dict['value'] = ' '.join(values_list[2:])
-        return values_dict
+        if len(values_list) <= 2:
+            raise Exception
+        return values_list
 
     def get_commands_dict(self) -> dict:
-        with open('./configuration/commands_dict.json') as f:
+        with open('./configuration/commands_dict.json') as f:  #todo: path to the file should be imported as parameter
             data = json.load(f)
         return data
 
-    def validate_operation(self, operation: str, commands_dict: dict, language: str) -> str:
-        #lngs_list = [language]
-        for key, value in commands_dict[language]['operation'].items():
-            if operation in value:
+    def get_data_by_language(self, language: str, data_dict: dict) -> dict:
+        if language in data_dict:
+            return data_dict[language]
+        else:
+            return data_dict['en-US']
+
+    def validate_word(self, word: str, possible_values: dict) -> str:
+        for key, value in possible_values.items():
+            if word in value:
                 return key
-        if language != 'en-US':
-            for key, value in commands_dict['en-US']['operation'].items():
-                if operation in value:
-                    return key
 
-    def prepare_command(self):
-        pass
+    def validate_command(self, words: list, possible_values: dict) -> dict:
+        _operation_list = []
+        _subject_list = []
+        _identifiers_list = []
+        _result = dict()
+        for word in words:
+            if self.validate_word(word, possible_values['operation']):
+                _operation_list.append(word)
+            elif self.validate_word(word, possible_values['subject']):
+                _subject_list.append(word)
+            else:
+                _identifiers_list.append(word)
+
+        if len(_operation_list) == 1 and len(_subject_list) == 1:
```

Comment:
I would prefer to fail earlier instead of doing block of operations and fail at the end.
Basically it means that your block could be rewritten to
```
if Len(_operation_list) != 1 or len(_subject_list) != 1:
   raise Exception(...)

normal block
```

Reply:
Done